src/utils/transactionUtils/transactionUtils.ts

```diff
@@ -588,3 +623,36 @@ export function validateTTLSeconds(ttl: number) {
     );
   }
 }
+
+export function getPendingInteractionsRowsForContract(
+  pendingContractInteractions: ContractInteraction[],
+  existingValues: any,
+): {
+  attribute: string;
+  value: string;
+  id: string;
+  valid: boolean | undefined;
+}[] {
+  // find all pending interactions for the contract, find relevant ones related to row attributes
+  const pendingTxRowData = [];
+  for (const i of pendingContractInteractions) {
+    const attributes = getAttributesFromInteractionFunction(i.payload.function);
+    // TODO: this is not pretty, and could be avoided if we rework the ANT contract to allow `setTTL` and `setTransaction` rather than all of them
+    // relying only on setRecord.
+    for (const attribute of attributes) {
+      const nonConfirmedTx = {
+        attribute,
+        value: i.payload[attribute],
```

Comment:
```suggestion
        value: i.payload[attribute === 'targetID' ? 'transactionId' : attribute],
```
We have a mismatching attribute and data key type - IMO the key in the contract should be renamed to 'targetId'. This is a bandaid fix for the moment ^ for the bug where a target ID edit didnt not show up.

Reply:
[27369b5](https://github.com/ar-io/pdns-react/pull/214/commits/27369b5c592e166ae1ea6dd9e9808e1d426eb027)